core/src/main/java/org/elasticsearch/index/percolator/PercolatorQueriesRegistry.java

```diff
@@ -184,14 +185,15 @@ Query parsePercolatorDocument(String id, BytesReference source) {
         }
     }
 
+    //norelease this method parses from xcontent to lucene query, need to re-investigate how to split context here
     private Query parseQuery(String type, XContentParser parser) {
```

Comment:
I think we will need to do the same that we are doing for searches, percolator queries will be parsed to our own query format when indexed, at execution time they will be converted to lucene query. This will come with a little cost, given that we currently parse directly to lucene, but it will also solve problems. Have a look at #12179 , maybe we can mention it here in the comment.